realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -154,11 +154,15 @@ public void addChangeListener(RealmChangeListener listener) {
      * Removes the specified change listener.
      *
      * @param listener the change listener to be removed.
+     * @throws IllegalStateException if you try to remove a listener from a non-Looper Thread.
      * @see io.realm.RealmChangeListener
      * @see #addChangeListener(RealmChangeListener)
      */
     public void removeChangeListener(RealmChangeListener listener) {
         checkIfValid();
+        if (handler == null) {
```

Comment:
Ah, yes, I got confused by this checking `handler` but below `handlerController` is used. It looks like both variables are guarded by the same `if`-statement. It would be more readable if this checked `handlerController` instead.

Otherwise, I can see why it makes sense to guard against it.